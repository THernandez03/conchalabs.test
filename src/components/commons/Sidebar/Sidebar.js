import { createPortal } from 'react-dom';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';

import { Overlay } from '../Overlay';

export const Sidebar = styled(({ className, children, isOpened }) => {
  if (!isOpened) return <div className={className} />;

  return (
    <div className={className}>
      {children}

      {createPortal(<Overlay />, document.querySelector('#modal-wrapper'))}
    </div>
  );
})`
  position: absolute;
  display: block;
  height: 100vh;
  background: #e8e8e8;
  width: 450px;
  z-index: 2;
  transition: left 0.5s;

  ${({ isOpened }) => `
    left: ${isOpened ? 0 : -450}px;
  `}

  @media (max-width: 550px) {
    width: 85%;
  }
`;

Sidebar.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  className: PropTypes.string,
  isOpened: PropTypes.bool,
};

Sidebar.defaultProps = {
  isOpened: false,
};
