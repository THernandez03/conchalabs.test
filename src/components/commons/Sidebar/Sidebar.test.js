import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';

import { Sidebar } from './Sidebar';

window.document.querySelector = jest.fn();

describe('<Sidebar/>', () => {
  it('Should render the component', () => {
    const createPortal = jest.spyOn(ReactDOM, 'createPortal');
    createPortal.mockImplementation(() => null);

    const { container } = render(<Sidebar isOpened />);

    expect(container).toMatchSnapshot();
    expect(createPortal).toHaveBeenCalled();
  });

  it('Should have different css when "isOpened" is true', () => {
    const { container, rerender } = render(<Sidebar isOpened />);
    expect(container.firstChild.className).toMatchSnapshot();
    rerender(<Sidebar />);
    expect(container.firstChild.className).toMatchSnapshot();
    rerender(<Sidebar isOpened />);
    expect(container.firstChild.className).toMatchSnapshot();
  });
});
