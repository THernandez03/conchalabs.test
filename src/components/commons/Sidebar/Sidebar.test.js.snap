// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar/> Should have different css when "isOpened" is true 1`] = `"css-1p6088f-Sidebar e19m2rzp0"`;

exports[`<Sidebar/> Should have different css when "isOpened" is true 2`] = `"css-t0ugvi-Sidebar e19m2rzp0"`;

exports[`<Sidebar/> Should have different css when "isOpened" is true 3`] = `"css-1p6088f-Sidebar e19m2rzp0"`;

exports[`<Sidebar/> Should render the component 1`] = `
<div>
  <div
    class="css-1p6088f-Sidebar e19m2rzp0"
  />
</div>
`;
